### PR TITLE
[strong-type] Update strong-type to v14

### PR DESCRIPTION
**Pull request by a4z**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/strong-type/portfile.cmake [hunk 1]
```diff
@@ -1,12 +1,12 @@
-vcpkg_from_github(OUT_SOURCE_PATH SOURCE_PATH
```
Comment (MonicaLiu0311) [inline ports/strong-type/portfile.cmake:1]: Please don't change the `EOL`, it will create unnecessary `Blame`.

**Comment by MonicaLiu0311** [comment]
At present, the version information of your commit is inconsistent with your local one:
```
##[error]Found the following errors:
##[error]versions/s-/strong-type.json(2,): error : while parsing versions for strong-type from version/s-/strong-type.json
14 is declared with 85f393f413a163ffc40bcecb73e423255339591f, but the local port has a different SHA dcd0430a39cb70a7e2e02263cd70f99d5f3f4ea1.
```

`--overwrite-version`
After the first modification, please run `./vcpkg x-add-version portName` to generate the corresponding json; for subsequent modifications of the same version, please use `./vcpkg x-add-version portName --overwrite-version` to update git-tree.

[maintainer guide#versioning](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#versioning).

**Comment by MonicaLiu0311** [comment]
The currently recommended modification steps are:
```
Restore the original EOL of portfile.cmake and commit the modifications.
Run `./vcpkg x-add-version portName --overwrite-version` to update git-tree and commit verison information again.
Push together.
```

**Comment by a4z** [comment]
Thanks for the help! I close this PR and continue there: https://github.com/microsoft/vcpkg/pull/37178

**Pull request closed without merging**